config/templates/elasticsearch-db.yaml

```diff
@@ -0,0 +1,105 @@
+apiVersion: templating.flanksource.com/v1
+kind: Template
+metadata:
+  name: elasticsearch-db
+spec:
+  source:
+    apiVersion: db.flanksource.com/v1
+    kind: ElasticsearchDB
+  resources:
+    - apiVersion: v1
+      kind: Namespace
+      metadata:
+        name: eck
+    - apiVersion: elasticsearch.k8s.elastic.co/v1
+      kind: Elasticsearch
+      metadata:
+        name: logs
+        namespace: eck
+      spec:
+        version: '{{ .spec.version }}'
+        nodeSets:
+          - name: default
+            count: '{{ .spec.replicas | default 3 }}'
+            config:
+              node.master: true
+              node.data: true
+              node.ingest: true
+              node.store.allow_mmap: false
+              xpack.security.transport.ssl.supported_protocols: TLSv1.1,TLSv1.2
+              xpack.security.authc.anonymous.roles: fluentd
+            podTemplate:
+              spec:
+                containers:
+                  - name: elasticsearch
+                    env:
+                      - name: ES_JAVA_OPTS
+                        value: '-Xms{{ .spec.mem.requests | default "2Gi" | strings.ReplaceAll "Ki" "k" | strings.ReplaceAll "Mi" "m" | strings.ReplaceAll "Gi" "g"  }} -Xmx{{ .spec.mem.requests | default "2Gi" | strings.ReplaceAll "Ki" "k" | strings.ReplaceAll "Mi" "m" | strings.ReplaceAll "Gi" "g" }}'
```

Comment:
This would probably need to be under .spec.HeapSize as .spec.Mem.Limit must be > HeapSize + 1 - 2Gi